### PR TITLE
Fix TestCollection.js

### LINKED ISSUE #11722: Update /pages/TestCollection.js to work with future dependencies
This one was somehow missed when I created all the other issues.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #11722

### CODE REVIEW

File: tests/unit/amo/pages/TestCollection.js [hunk 1]
```diff
@@ -1019,6 +1021,10 @@ describe(__filename, () => {
       }),
     });
 
+    const dispatch = jest
```
Comment (bobsilverberg) [inline tests/unit/amo/pages/TestCollection.js:1024]: This is what I had to do to resolve the problem of the tests running for a very long time and eventually resulting in a JS oom error. I was never able to fully debug it and trace everything through, but it seems like allowing the server redirect to be fired in the test was causing the issue. Perhaps, for some reason, it was causing an infinite loop? We have other tests where we are checking something similar, and they don't seem to have any issues, so this is a bit weird. I tried adding debugging info, but it just stopped being output at some point. It's probably worth doing a bit more investigation into this, but for now this seems like a reasonable workaround.

Comment (diox) [reply tests/unit/amo/pages/TestCollection.js:1024]: Just in case we should flag that for QA to double-check once the dependency updates go live.

Comment (bobsilverberg) [reply tests/unit/amo/pages/TestCollection.js:1024]: Good idea!

**Review by diox — APPROVED** [review]

File: tests/unit/amo/pages/TestCollection.js (hunk 1 above)
Comment (bobsilverberg) [reply tests/unit/amo/pages/TestCollection.js:1024]: I added a comment to https://github.com/mozilla/addons-frontend/pull/11652#issue-1294644653

File: tests/unit/amo/pages/TestCollection.js [hunk 2]
```diff
@@ -2056,10 +2077,8 @@ describe(__filename, () => {
     });
 
     it('calls history.push() when creating', () => {
-      const history = createHistory({
-        initialEntries: [`/${lang}/${clientApp}/collections/add/`],
-      });
-      renderInAddMode({ history });
+      // renderInAddMode({ location: `/${lang}/${clientApp}/collections/add/` });
```
[the remark this answers is not in the document]
Comment (bobsilverberg) [reply tests/unit/amo/pages/TestCollection.js:2080]: No, that was just me commenting out that code, and forgetting to delete it.

**Pull request merged**